hudi-common/src/main/avro/HoodieRequestedReplaceMetadata.avsc

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+{
+   "namespace":"org.apache.hudi.avro.model",
```

Comment:
Same comment, just roll up all the datastructures that you need into the HoodieClusteringPlan, is it possible to merge the clustering plan with the insert-overwrite plan ?

Reply:
I think this model is flexible for future operations. Replace could be used for other operations such as column pruning and many other use cases. Trying to fit metadata for all these special operations into one format may cause problems (InsertOverwrite doesnt really have a plan today.  so we have to generate plan just for sake of creating it). We also dont want to introduce other top level commits because its lot of work, so i want to keep this as generic as possible.

Reply:
Keeping the model flexible sounds fair to me, I'm not following how that is connected to introducing other top level commits, could you please expand ?

Reply:
initially we were considering replacecommit, clusteringcommit as separate things. Only reason not to introduce new commit is to keep it simple as post commit metadata looks very similar. I was trying to say that if these were separate top level commits, this file is not needed.